src/main/java/seedu/tasklist/storage/JsonAdaptedLesson.java

```diff
@@ -55,17 +55,17 @@ public Lesson toModelType() throws IllegalValueException {
 
         if (startTime == null) {
             throw new IllegalValueException(
-                    String.format(MISSING_FIELD_MESSAGE_FORMAT, Deadline.class.getSimpleName()));
+                    String.format(MISSING_FIELD_MESSAGE_FORMAT, Time.class.getSimpleName()));
         }
-        if (!Deadline.isValidDeadline(startTime)) {
-            throw new IllegalValueException(Deadline.MESSAGE_CONSTRAINTS);
+        if (!Time.isValidDeadline(startTime)) {
+            throw new IllegalValueException(Time.MESSAGE_CONSTRAINTS);
         }
-        final Deadline modelStartTime = new Deadline(startTime);
+        final Time modelStartTime = new Time(startTime);
 
-        if (!Deadline.isValidDeadline(endTime)) {
-            throw new IllegalValueException(Deadline.MESSAGE_CONSTRAINTS);
+        if (!Time.isValidDeadline(endTime)) {
+            throw new IllegalValueException(Time.MESSAGE_CONSTRAINTS);
```

Comment:
Follow up to my previous comment on the naming of isValidDeadline, since isValidDeadline is used in to check the format for Lesson as well, it would be good to rename the method to isValidTime